### PR TITLE
Add make completion out of box

### PR DESCRIPTION
**Pull request by fwfurtado**
# Description
Add target completions from _Makefile_ in _default_config.nu_.

# User-Facing Changes

N/A

# Tests + Formatting

Don't forget to add tests that cover your changes.

Make sure you've run and fixed any issues with these commands:

- `cargo fmt --all -- --check` to check standard code formatting (`cargo fmt --all` applies these changes)
- `cargo clippy --workspace --features=extra -- -D warnings -D clippy::unwrap_used -A clippy::needless_collect` to check that you're using the standard code style
- `cargo test --workspace --features=extra` to check that all tests pass

# After Submitting

If your PR had any user-facing changes, update [the documentation](https://github.com/nushell/nushell.github.io) after the PR is merged, if necessary. This will help us keep the docs up to date.

### CODE REVIEW

File: crates/nu-utils/src/sample_config/default_config.nu [hunk 1]
```diff
@@ -122,6 +122,22 @@ module completions {
     --verbose(-v)                                   # be more verbose
     --help                                          # Display the help message for this command
   ]
+
+  # This is a simplified version of completions for make targets in a Makefile
+  def 'nu-complete make targets' [] {
+    if ("Makefile" | path exists) {
+      open Makefile | lines | where ($it | $it =~ '^\S+:') | each { |it| $it | str substring  0..($it | str index-of  ':')  }
```
[the remark this answers is not in the document]
Comment (fwfurtado) [reply crates/nu-utils/src/sample_config/default_config.nu]: I've fixed to ignore commented lines.
But if I'm not mistaken, this `srcdir_b2h=		C:/M/mingw-w64-python/src/Python-3.10.8` is not a make target, right?

[the remark this answers is not in the document]
Comment (fwfurtado) [reply crates/nu-utils/src/sample_config/default_config.nu]: I'm testing with this _Makefile_:
```
srcdir_b2h=C:/M/mingw-w64-python/src/Python-3.10.8

.PHONY: list
#commented/out:
#	@ echo ""

uncommented/line:
	@ echo "srcdir_b2h=		$(srcdir_b2h)"

a:
	@ echo "A"

%.c:
	@ echo "% c"

%.o : %.c
	@ echo "% o"

# %:
# 	@ echo "%"

foo bar biz &:
	@echo grouped target run for \"target\" $@

kbd.o command.o files.o: command.h

file-a%out file-b%out:
	@ echo "files"
```

Is there any way to create an unit test for _default_config.nu_?

**Comment by webbedspace** [comment]
FWIW I pulled up a random Makefile from some directory as an example; if you want the full file for testing it's here:
[Makefile.txt](https://github.com/nushell/nushell/files/10073459/Makefile.txt)

**Comment by sholderbach** [comment]
I don't want to ruin the party here, having accurate dynamic make completions is absolutely crucial for a good nushell experience.
But I think the declared goal for the `default_config.nu` file should be that it describes the core functionality and you should be able to import the things you need. (In that sense the few git completions are doing more harm then good as we don't really move forward shipping a best practice architecture importing more completions)

We have community developed completion definitions in [nu_scripts](https://github.com/nushell/nu_scripts) (e.g. [someone already started implementing the flags for make](https://github.com/nushell/nu_scripts/blob/main/custom-completions/make/make-completions.nu)

**Comment by kubouch** [comment]
I agree, feel free to submit the completions to the nu_scripts repo but I wouldn't clutter the default config. The current ones there are more like an example (and need to be trimmed down IMO).

**Pull request closed without merging**